tools/clang/unittests/HLSL/DxilResourceTests.cpp

```diff
@@ -0,0 +1,67 @@
+//===- unittests/DXIL/DxilResourceTests.cpp ----- Tests for DXIL Resource -===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "dxc/DXIL/DxilConstants.h"
+#include "dxc/Support/WinIncludes.h"
+
+#include "dxc/Test/HlslTestUtils.h"
+
+using namespace hlsl;
+using namespace hlsl::DXIL;
+
+#ifdef _WIN32
+class DxilResourceTest {
+#else
+class DxilResourceTest : public ::testing::Test {
+#endif
+public:
+  BEGIN_TEST_CLASS(DxilResourceTest)
+  TEST_CLASS_PROPERTY(L"Parallel", L"true")
+  TEST_METHOD_PROPERTY(L"Priority", L"0")
+  END_TEST_CLASS()
+
+  TEST_METHOD(KindChecks)
+};
+
+struct KindTestCase {
+  ResourceKind Kind;
+  bool IsTexture;
+  bool IsTextureArray;
+  bool IsTextureCube;
+  bool IsFeedbackTexture;
+};
+
+KindTestCase Kinds[(int)ResourceKind::NumEntries] = {
+    {ResourceKind::Texture1D, true, false, false, false},
+    {ResourceKind::Texture2D, true, false, false, false},
+    {ResourceKind::Texture2DMS, true, false, false, false},
+    {ResourceKind::Texture3D, true, false, false, false},
+    {ResourceKind::TextureCube, true, false, true, false},
+    {ResourceKind::Texture1DArray, true, true, false, false},
+    {ResourceKind::Texture2DArray, true, true, false, false},
+    {ResourceKind::Texture2DMSArray, true, true, false, false},
+    {ResourceKind::TextureCubeArray, true, true, true, false},
+    {ResourceKind::TypedBuffer, false, false, false, false},
+    {ResourceKind::RawBuffer, false, false, false, false},
+    {ResourceKind::StructuredBuffer, false, false, false, false},
+    {ResourceKind::CBuffer, false, false, false, false},
+    {ResourceKind::Sampler, false, false, false, false},
+    {ResourceKind::TBuffer, false, false, false, false},
+    {ResourceKind::RTAccelerationStructure, false, false, false, false},
+    {ResourceKind::FeedbackTexture2D, false, false, false, true},
+    {ResourceKind::FeedbackTexture2DArray, false, false, false, true}};
```

Comment:
Since `FeedbackTexture2d` was excluded from `IsAnyTexture` I assumed `FeedbackTexture2DArray` should also be excluded from `IsAnyTextureArray`. Not sure if that is the correct assumption, but it is what I went with.

Reply:
@tex3d what do you think about this? Should a FeedbackTexture2d be considered a texture (and similarly for the array case)?

Reply:
I think I agree that `FeedbackTexture2DArray` shouldn't be considered a texture or array texture.  It's really the feedback buffer for a Texture2DArray, not the Texture2DArray itself.

Reply:
Thanks @tex3d!

If there's no other concerns I'll go a head and merge this :)

Reply:
LGTM!

Reply:
Though I haven't given it a ton of thought yet, so I could be convinced otherwise ;)

Reply:
Looks like we are going to change to `IsAnyArray` to cover feedback array property.  However, I also realized that this test array could use a check to make sure all resources are in the table here.  For instance, declaring the array with `[]` and adding a static assert that the array size matches the `ResourceKind::NumEntries`.  That way, the static assert will cause compile failure when a kind is added, but the test doesn't get updated.